src/test/java/seedu/placebook/logic/commands/FindAppCommandTest.java

```diff
@@ -0,0 +1,110 @@
+
+    @Test
+    public void equals() {
+        DescriptionContainsKeywordsPredicate firstPredicate =
+                new DescriptionContainsKeywordsPredicate(Collections.singletonList("first"));
+        DescriptionContainsKeywordsPredicate secondPredicate =
+                new DescriptionContainsKeywordsPredicate(Collections.singletonList("second"));
+
+        FindAppCommand findFirstCommand = new FindAppCommand(firstPredicate);
+        FindAppCommand findSecondCommand = new FindAppCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindAppCommand findFirstCommandCopy = new FindAppCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different appointment -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_zeroKeywords_noAppointmentFound() {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 0);
+        DescriptionContainsKeywordsPredicate predicate = preparePredicate(" ");
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, uiStub, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_oneKeyword_multipleAppointmentsFound() {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 2);
+        DescriptionContainsKeywordsPredicate predicate = preparePredicate("sales");
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, uiStub, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE_APPOINTMENT, CARL_APPOINTMENT), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_oneKeyword_oneAppointmentFound() {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 1);
+        DescriptionContainsKeywordsPredicate predicate = preparePredicate("talk");
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, uiStub, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE_APPOINTMENT), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_multipleAppointmentsFound() {
```

Comment:
Should we test the case where theres multiple keywords  but only single appointment found

Reply:
thanks for your comment! 
I refined the logic for my test, so the three test are
keyword | results
----------|---------
`sales` | 2
`talk` | 1
`sales` `talk` | 2 

so the logic is trying to see that the searching works like a `or` logic.